invoice.py

```diff
@@ -265,19 +265,58 @@ def get_intercompany_account(self):
     def get_intercompany_taxes(self):
         pool = Pool()
         Product = pool.get('product.product')
-        taxes = []
+
+        taxes = set()
         if not self.product:
             return taxes
+
         target_company = self.invoice.target_company
-        type = self.invoice.type if self.invoice else self.invoice_type
-        tax_name = '%s_taxes_used' % ('customer' if type[:2] == 'in'
-            else 'supplier')
+        party = target_company.party
+        type_ = self.invoice.type if self.invoice else self.invoice_type
 
         with Transaction().set_user(target_company.intercompany_user.id), \
                 Transaction().set_context(company=target_company.id,
                 _check_access=False):
             product = Product(self.product.id)
-            taxes = getattr(product, tax_name, [])
+
+            if type_ == 'out': # if is out, taxes from supplier (in)
+                tax_rule = party.supplier_tax_rule
+                tax_rule_template = (hasattr(target_company, 'supplier_tax_rule_template')
+                    and getattr(target_company, 'supplier_tax_rule_template'))
+                if tax_rule_template:
+                    tax_rule = tax_rule_template.get_syncronized_company_value(target_company)
+
+                pattern = self._get_tax_rule_pattern()
+                for tax in product.supplier_taxes_used:
+                    if party and tax_rule:
+                        tax_ids = tax_rule.apply(tax, pattern)
+                        if tax_ids:
+                            taxes.update(tax_ids)
+                        continue
+                    taxes.add(tax.id)
+                if party and tax_rule:
+                    tax_ids = tax_rule.apply(None, pattern)
+                    if tax_ids:
+                        taxes.update(tax_ids)
+            else:
+                tax_rule = party.customer_tax_rule
+                tax_rule_template = (hasattr(target_company, 'supplier_tax_rule_template')
+                    and getattr(target_company, 'supplier_tax_rule_template'))
```

Comment:
Is "supplier_tax_rule_template" correct here?

Reply:
FIX 103cabe9ef3ea1ba9b79d67c6293ffb2f210e6c2